### ISSUE TITLE
Ionic2 push notifications not showing when app close

### ISSUE THREAD

**Issue by jdileep14** [arnesson/cordova-plugin-firebase#482]
Hi i am using firebase for push notifications . I am getting push notification when open or background but push notifications are not showing when app close .It is unsigned app .Please help me thanks in advance .

**Comment by peterlai107** [comment]
What is your platform? and how do you send your message?

**Comment by jdileep14** [comment]
this is my ionic info
cli packages: (/usr/local/lib/node_modules)

    @ionic/cli-utils  : 1.18.0
    ionic (Ionic CLI) : 3.18.0

global packages:

    cordova (Cordova CLI) : 7.1.0 

local packages:

    @ionic/app-scripts : 3.0.0
    Cordova Platforms  : android 6.3.0 ios 4.5.2
    Ionic Framework    : ionic-angular 3.7.1

System:

    Node  : v6.9.0
    npm   : 5.5.1 
    OS    : macOS Sierra
    Xcode : Xcode 8.3.3 Build version 8E3004b 

Environment Variables:

    ANDROID_HOME : not set

Misc:

    backend : pro


basically i want to send push notification if app close. using firbase i am getting notifications when app  open or in background but i am not getting push notification when app close

**Comment by peterlai107** [comment]
I saw you using xcode, so I assume your platform is iOS.
There are two types of messages (notification type / data type).
Which message type do you send? 

If you just want receiving notification only, my experience is as below
If you want receiving notification and wake up your app(maybe want to execute some code) in a forced close state, that is hard to achieve

|ios|notification|data|
|---|---|---|
|foreground|v|v|
|background|v|v|
|force close|v|x|

Refer to this doc [firebase doc](https://firebase.google.com/docs/cloud-messaging/http-server-ref)


|content_available|Optional, boolean|On iOS, use this field to represent content-available in the APNs payload. When a notification or message is sent and this is set to true, an inactive client app is awoken, and the message is sent through APNs as a silent notification and not through the FCM connection server. Note that silent notifications in APNs are not guaranteed to be delivered, and can depend on factors such as the user turning on Low Power Mode, force quitting the app, etc. On Android, data messages wake the app by default. On Chrome, currently not supported.|
|---|---|---|

Hope this helps.
